var/spack/repos/builtin/packages/xtensor-python/package.py

```diff
@@ -16,11 +16,14 @@ class XtensorPython(CMakePackage):
     maintainers = ['ax3l']
 
     version('develop', branch='master')
+    version('0.23.1', sha256='450b25f5c739df174b2a50774b89e68b23535fdc37cb55bd542ffdb7c78991ab')
     version('0.17.0', '51d22e42909a81201c3421d9e119eed0')
 
-    depends_on('xtensor@0.15.1:0.15.99', when='@0.17.0:')
-    depends_on('xtl@0.4.0:0.4.99', when='@0.17.0:')
-    depends_on('py-pybind11@2.2.1', when='@0.17.0:')
+    depends_on('xtensor@0.20.6:0.20.99', when='@0.23.1')
+    depends_on('xtensor@0.15.1:0.15.99', when='@0.17.0')
+    depends_on('xtl@0.6.4:0.6.99', when='@0.23.1')
+    depends_on('xtl@0.4.0:0.4.99', when='@0.17.0')
```

Comment:
What about `develop`? I generally prefer open version ranges if you have the time to determine exactly when the update occurred.

Reply:
Thanks for catching this! `develop` just depends on `xtl` latest, updated now accordingly.

Ignoring that major releases are likely breaking is imho a source of constant build breakage which I deal with daily in user reports downstream. Luckily, `xtensor` uses the [`^`/caret notation](https://nodesource.com/blog/semver-tilde-and-caret/) for versions which also only supports the same major: https://github.com/QuantStack/xtensor#dependencies

Reply:
What do you mean when you say "latest"? As is, you're saying that `xtensor-python` can be built with _any_ version of `xtensor` and `xtl`, even very old versions. When I said I prefer open version ranges, I meant in the `when` argument. For example, something like:
```python
depends_on('xtensor')
depends_on('xtensor@0.20.6:0.20.99', when='@0.22:')
depends_on('xtensor@0.15.1:0.15.99', when='@:0.21')
```
assuming that 0.22 was the first release where this version requirement was updated.

Reply:
Yes, but the dependencies in `xtensor` and `xtensor-python` state a minimum version and an ["up to next major yet not including it"](https://nodesource.com/blog/semver-tilde-and-caret/):
- https://github.com/QuantStack/xtensor#dependencies
- https://github.com/QuantStack/xtensor-python#dependencies-on-xtensor-and-pybind11

For `develop` I just keep the range undefined since spack will default on the last checked in version.

Reply:
I agree with Adam that it may be confusing to leave the dependency constraint off of the `develop` versions because strictly speaking it allows the `develop` version of `xtensor-python` to build with any version (including old versions) of `xtensor`.

Pragmatically though: this will always prefer the latest xtensor that is available (provided that `packages.yaml` doesn't name an old version as preferred), and as time goes on, the minimum version of `xtensor` that is required will change, so a constraint like `depends_on(xtensor@0.20.6:, when=@develop)` may become "out of date".